### ISSUE TITLE
Node v8.5.0 breaks the "uncaught exception" setting

### ISSUE THREAD

**Issue by samrg472** [Microsoft/vscode#34593]
- VSCode Version: 1.17.0-insider (b0c0634)
- OS Version: MacOS 10.13 Beta (17A362a)
- Node Version: 8.5.0

Using the same sample code from #1746
```ts
function failingFunc() {
  return (undefined as any).nope();
}

let promise = new Promise((resolve, reject) => {
 resolve(failingFunc());
});

promise.then(v => {}).catch((e: Error) => {
  console.log(e.message);
});
```

"Uncaught exceptions" is disabled, however it keeps breaking, regardless if it checked or unchecked.

<img width="333" alt="screen shot 2017-09-18 at 2 39 48 pm" src="https://user-images.githubusercontent.com/903556/30566117-91b2f20a-9c7f-11e7-8f1b-35ff09f65ede.png">

Reproduces without extensions: Yes

**Comment by samrg472** [comment]
I was able to confirm this in the stable version of VSCode as well. I checked whether node was the cause and it looks like it is. Switching to `node v8.4.0` no longer has this issue.

**Comment by roblourens** [comment]
I can repro as well, with something as simple as 

```js
console.log('before');
Promise.reject(new Error('error'))
```

This is a Node bug, I filed https://github.com/nodejs/node/issues/15464

**Comment by roblourens** [comment]
You can workaround it in VS Code with

```
"skipFiles": [ "<node_internals>/**" ]
```

in your launch config

**Comment by MattMorgis** [comment]
The fix for this landed in [`be2a5b3`](https://github.com/nodejs/node/commit/be2a5b323e331aaef2991c7d7004f2ec833123f0) and will be released in Node v8.7. 

The corresponding [nodejs/#15464](https://github.com/nodejs/node/issues/15464) is closed, so probably safe to close this too? Just wanted to help clean up 👍
